File: lib/models/people_model.dart
// ignore_for_file: public_member_api_docs, sort_constructors_first
import 'dart:convert';

import 'package:json_annotation/json_annotation.dart';

part 'people_model.g.dart';

@JsonSerializable()
class PeopleModel {
  String? nationID;
  String? titleTH;
  String? firstnameTH;
  String? lastnameTH;
  String? titleEN;
  String? firstnameEN;
  String? lastnameEN;
  String? address;
  String? birthdate;
  String? issueDate;
  String? expireDate;
  int? gender;
  String? photo; //base64
  String? mobile;

  PeopleModel({
    this.nationID,
    this.titleTH,
    this.firstnameTH,
    this.lastnameTH,
    this.titleEN,
    this.firstnameEN,
    this.lastnameEN,
    this.address,
    this.birthdate,
    this.issueDate,
    this.expireDate,
    this.gender,
    this.photo,
    this.mobile,
  });

  factory PeopleModel.fromJson(Map<String, dynamic> json) =>
      _$PeopleModelFromJson(json);
  Map<String, dynamic> toJson() => _$PeopleModelToJson(this);

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'nationID': nationID,
      'titleTH': titleTH,
      'firstnameTH': firstnameTH,
      'lastnameTH': lastnameTH,
      'titleEN': titleEN,
      'firstnameEN': firstnameEN,
      'lastnameEN': lastnameEN,
      'address': address,
      'birthdate': birthdate,
      'issueDate': issueDate,
      'expireDate': expireDate,
      'gender': gender,
      'photo': photo,
      'mobile': mobile,
    };
  }
}

File: server/internal/api/v1/people/model/people.model.go
package model

// {
//     "user_nationID": "123xxxxxxxxxx",
//     "personData": {
//         "nationID": "141xxxxxxxxxx",
//         "titleTH": "น.ส.",
//         "firstnameTH": "กุล...",
//         "lastnameTH": "ปะ....",
//         "titleEN": "Miss",
//         "firstnameEN": "Kuxxxxxx",
//         "lastnameEN": "Paxxxxx",
//         "address": "xxx xxxxx xxxxxx xxxxx",
//         "birthdate": "xxxx-xx-xx",
//         "issueDate": "xxxx-xx-xx",
//         "expireDate": "xxxx-xx-xx",
//         "gender": 2,
//         "photo": "base64",
//         "mobile": "093389xxxx"
//     },
//     "location": {
//         "latitude": 17.0761887,
//         "longitude": 102.9304177
//     }
// }

type CreatePeopleDTO struct {
	UserNationID string     `json:"userNationID" validate:"required"`
	PeopleData   PeopleData `json:"personData" validate:"required"`
	Location     Location   `json:"location" validate:"required"`
}

type PeopleData struct {
	NationID    string `json:"nationID" validate:"required"`
	TitleTH     string `json:"titleTH" validate:"required"`
	FirstnameTH string `json:"firstnameTH" validate:"required"`
	LastnameTH  string `json:"lastnameTH" validate:"required"`
	TitleEN     string `json:"titleEN" validate:"required"`
	FirstnameEN string `json:"firstnameEN" validate:"required"`
	LastnameEN  string `json:"lastnameEN" validate:"required"`
	Address     string `json:"address" validate:"required"`
	Birthdate   string `json:"birthdate" validate:"required"`
	IssueDate   string `json:"issueDate" validate:"required"`
	ExpireDate  string `json:"expireDate" validate:"required"`
	Gender      int    `json:"gender" validate:"required"`
	Photo       string `json:"photo" validate:"omitempty"`
	Mobile      string `json:"mobile" validate:"required"`
}

type Location struct {
	Latitude  float64 `json:"latitude" validate:"required"`
	Longitude float64 `json:"longitude" validate:"required"`
}

type ImageDTO struct {
	OriginName string `json:"name" validate:"required,min=5,max=100"`
	Extension  string `json:"extension" validate:"required"`
	Reference  string `json:"reference" validate:"required"`
}
